modules/balance-ignored-labels-autoscaler-hcp.adoc

```diff
@@ -0,0 +1,114 @@
++
+[source,terminal]
+----
+$ oc patch -n <hosted_cluster_namespace> \
+  hostedcluster<hosted_cluster_name> \
+  --type=merge \
+  --patch='{"spec": {"autoscaling": {"scaling": "ScaleUpAndScaleDown", "maxPodGracePeriod": 60, "maxFreeDifferenceRatioPercent": 50, "scaleDown": {"delayAfterAddSeconds": 60, "delayAfterDeleteSeconds": 60, "delayAfterFailureSeconds": 60, "unneededDurationSeconds": 60, "utilizationThresholdPercent": 50}, "expanders": ["Random"]}}}'
+----
+
+. Remove the `spec.replicas` field from the `NodePool` resource so that cluster autoscaler can manage the node count. Run the following command:
++
+[source,terminal]
+----
+$ oc patch -n <hosted_cluster_namespace> \
+  nodepool <node_pool_name> \
+  --type=json \
+  --patch='[{"op": "remove", "path": "/spec/replicas"}]'
+----
+
+. Generate the `kubeconfig` file by running the following command:
++
+[source,terminal]
+----
+$ hcp create kubeconfig \
+  --name <hosted_cluster_name> \
+  --namespace <hosted_cluster_namespace> > nested.config
+----
+
+. After the node pool scales up, check that all worker nodes are in the `Ready` status by running the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig nested.config get nodes -l 'hypershift.openshift.io/nodePool=<node_pool_name>'
+----
+
+. Ensure that the newly provisioned nodes contain the `node.group.balancing.ignored` label by running the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig nested.config get nodes \
+  -l 'hypershift.openshift.io/nodePool=<node_pool_name>' \
+  -o jsonpath='{.items[*].metadata.labels}' | grep "node.group.balancing.ignored"
+----
+
+. Ensure that the cluster autoscaler pod does not yet contain the `--balancing-ignore-label=node.group.balancing.ignored` argument by running the following command:
++
+[source,terminal]
+----
+$ oc get pod <cluster_autoscaler_pod_name> -n <hosted_control_plane_namespace> \
+  -o jsonpath='{.spec.containers[0].args}'
+----
+
+. Enable cluster autoscaling for your hosted cluster to ignore the balancing label by running the following command:
++
+[source,terminal]
+----
+$ oc patch -n <hosted_cluster_namespace> \
+  hostedcluster <hosted_cluster_name> \
+  --type=merge \
+  --patch='{"spec": {"autoscaling": {"balancingIgnoredLabels": ["node.group.balancing.ignored"]}}}'
+----
+
+.Verification
+
+. Verify that the new nodes contain the `node.group.balancing.ignored` label by running the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig nested.config get nodes \
+  -l 'hypershift.openshift.io/nodePool=<node_pool_name>' \
+  -o jsonpath='{.items[*].metadata.labels}' | grep "node.group.balancing.ignored"
+----
+
+. Verify that the cluster autoscaler pod contains the `--balancing-ignore-label=node.group.balancing.ignored` argument by running the following command:
++
+[source,terminal]
+----
+$ oc get pod <cluster_autoscaler_pod_name> \
```

Comment:
I am not sure, but maybe we can ignore the part of the autoscaler pod arguments verification, it is internal used for debug purpose.